### ISSUE TITLE
[Feature] Add Structured Metric Logging API for Step-wise Experiment Tracking

### ISSUE THREAD

**Issue by kaurroopak** [BBEK-Anand/PyLabFlow#13]
**Description**
PyLabFlow is designed to support reproducible computational workflows, including machine learning experiments. However, there is currently no structured mechanism for logging step-wise metrics (e.g., training loss, validation accuracy, etc.) during workflow execution.
For ML and iterative experiments, tracking metrics across epochs/steps is essential for:

- Monitoring training progress
- Comparing pipelines quantitatively
- Debugging model performance
- Ensuring reproducibility of reported results
Currently, the status() method returns a dictionary, but it does not appear to support structured historical metric storage.

**Proposed Enhancement**

Introduce a structured metric logging API within WorkFlow, for example:
self.log_metric(name: str, value: float, step: int)

**Suggested behavior:**

- Metrics are automatically associated with the current pplid
- Stored persistently (e.g., in SQLite database)
- Support multiple metrics per step
- Allow retrieval via existing experiment management tools (e.g., get_ppl_status() or a new get_metrics(pplid) function)

**Possible Implementation Ideas**

- Create a dedicated metrics table in the SQLite database:

    - pplid
    - metric_name
    - value
    - step
    - timestamp

- Expose retrieval utilities for:

    - Fetching full metric history
    - Aggregating latest values per metric
    - Comparing metrics across pipelines

**Benefits**

- Makes PyLabFlow more suitable for ML and deep learning workflows
- Enhances experiment comparison capabilities
- Strengthens reproducibility claims by preserving full metric history
- Keeps the framework aligned with offline-first principles

@BBEK-Anand I would be happy to work on this feature if it aligns with the project roadmap.

**Comment by BBEK-Anand** [comment]
Hi @kaurroopak 

we have 2 example setups, mentioned in [discussion ](https://github.com/BBEK-Anand/PyLabFlow/discussions/2)
ML Setup:
👉 https://github.com/ExperQuick/PLF_SetUps/tree/main/PLFML

Basic DL Setup:
👉 https://github.com/ExperQuick/PLF_DL_SetUps/tree/main/Basic

go through them once.

for flexibility, the logging  should be implemented by the researchers only, as we are domain independent.
